### ISSUE TITLE
Updater not set resources request on pods

### ISSUE THREAD

**Issue by qw1mb0** [kubernetes/autoscaler#1588]
I installed VPA 0.3.0 in a cluster 1.10.8 on two nodes and a master in GCE.

Have enabled MutatingAdmissionWebhook:
```
# kubectl -n kube-system get pod kube-apiserver-master-europe-west1-b-l6x1 -o yaml | grep MutatingAdmissionWebhook
      --cloud-provider=gce --enable-admission-plugins=Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,NodeRestriction,ResourceQuota
```
and enabled api:
```
# kubectl api-versions | grep admissionregistration.k8s.io
admissionregistration.k8s.io/v1beta1
```

And I wanted to check his work by applying such yaml:
```
apiVersion: "autoscaling.k8s.io/v1beta1"
kind: VerticalPodAutoscaler
metadata:
  name: hamster-vpa
spec:
  selector:
    matchLabels:
      app: hamster
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: hamster
spec:
  replicas: 2
  template:
    metadata:
      labels:
        app: hamster
    spec:
      containers:
      - name: hamster
        image: k8s.gcr.io/ubuntu-slim:0.1
        resources:
          requests:
            cpu: 100m
            memory: 50Mi
          limits:
            cpu: 300m
            memory: 250Mi
        command: ["/bin/sh"]
        args: ["-c", "while true; do timeout 0.5s yes >/dev/null; sleep 0.5s; done"]
```

It's deployed to cluster:
```
# kubectl get vpa,pod
NAME                                                   AGE
verticalpodautoscaler.autoscaling.k8s.io/hamster-vpa   3h

NAME                           READY     STATUS    RESTARTS   AGE
pod/hamster-5c6f76c9bb-4lbwt   1/1       Running   0          2m
pod/hamster-5c6f76c9bb-h7tqm   1/1       Running   0          37s
```

Every 1m one of pod recreate and I see on **Updater** logs:

```
I0116 23:01:37.161863       5 update_priority_calculator.go:118] pod accepted for update hamster-5c6f76c9bb-h7tqm with priority 6
I0116 23:01:37.162488       5 update_priority_calculator.go:118] pod accepted for update hamster-5c6f76c9bb-4lbwt with priority 6
I0116 23:01:37.162567       5 updater.go:137] evicting pod hamster-5c6f76c9bb-h7tqm
I0116 23:01:37.270985       5 updater.go:195] Event(v1.ObjectReference{Kind:"Pod", Namespace:"default", Name:"hamster-5c6f76c9bb-h7tqm", UID:"898279de-19e2-11e9-8af9-42010a840006", APIVersion:"v1", ResourceVersion:"32694081", FieldPath:""}): type: 'Normal' reason: 'EvictedByVPA' Pod was evicted by VPA Updater to apply resource recommendation.
```

See that vpa can fetch metrics seen on VPA object:
```
# kubectl describe vpa/hamster-vpa
Name:         hamster-vpa
Namespace:    default
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"autoscaling.k8s.io/v1beta1","kind":"VerticalPodAutoscaler","metadata":{"annotations":{},"name":"hamster-vpa","namespace":"default"},"spe...
API Version:  autoscaling.k8s.io/v1beta1
Kind:         VerticalPodAutoscaler
Metadata:
  Cluster Name:        
  Creation Timestamp:  2019-01-16T19:57:50Z
  Generation:          1
  Resource Version:    32694279
  Self Link:           /apis/autoscaling.k8s.io/v1beta1/namespaces/default/verticalpodautoscalers/hamster-vpa
  UID:                 011af3da-19c9-11e9-8af9-42010a840006
Spec:
  Selector:
    Match Labels:
      App:  hamster
Status:
  Conditions:
    Last Transition Time:  2019-01-16T19:59:11Z
    Status:                True
    Type:                  RecommendationProvided
  Recommendation:
    Container Recommendations:
      Container Name:  hamster
      Lower Bound:
        Cpu:     503m
        Memory:  262144k
      Target:
        Cpu:     671m
        Memory:  262144k
      Uncapped Target:
        Cpu:     671m
        Memory:  262144k
      Upper Bound:
        Cpu:     6200m
        Memory:  262144k
Events:          <none>

```

But every new pod have old resource request: 

```
# kubectl get pod
kNAME                       READY     STATUS    RESTARTS   AGE
hamster-5c6f76c9bb-4lbwt   1/1       Running   0          4m
hamster-5c6f76c9bb-7rmcz   1/1       Running   0          46s

# kubectl describe pod hamster-5c6f76c9bb-7rmcz
Name:           hamster-5c6f76c9bb-7rmcz
Namespace:      default
Node:           nodes-z27n/10.132.0.5
Start Time:     Wed, 16 Jan 2019 23:02:37 +0000
Labels:         app=hamster
                pod-template-hash=1729327566
Annotations:    <none>
Status:         Running
IP:             100.96.2.97
Controlled By:  ReplicaSet/hamster-5c6f76c9bb
Containers:
  hamster:
    Container ID:  docker://140bf6968b8cd041bea1587f0daf9a245e211d60b635d347fbcb038dade85429
    Image:         k8s.gcr.io/ubuntu-slim:0.1
    Image ID:      docker-pullable://k8s.gcr.io/ubuntu-slim@sha256:b6f8c3885f5880a4f1a7cf717c07242eb4858fdd5a84b5ffe35b1cf680ea17b1
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
    Args:
      -c
      while true; do timeout 0.5s yes >/dev/null; sleep 0.5s; done
    State:          Running
      Started:      Wed, 16 Jan 2019 23:02:38 +0000
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     300m
      memory:  250Mi
    Requests:
      cpu:        100m
      memory:     50Mi
```

And i not see some annotations on these pods.

Why VPA not working for me? Helm please.

**Comment by bskiba** [comment]
Do you have logs from admission controller? If the pods are restarted but not modified, there is something wrong with vpa-admission-controller.
`kubectl get logs -n kube-system vpa-admission-controller`

Can you also paste output of `kubectl describe mutatingWebhookConfiguration vpa-webhook-config`? This will tell us if the webhook is correctly registered with the API server.

**Comment by qw1mb0** [comment]
Logs for admission-controller:

```
# kubectl -n kube-prometheus logs vpa-admission-controller-79865cc446-zztb2
I0117 09:25:51.722796       1 flags.go:52] FLAG: --address=":8944"
I0117 09:25:51.723735       1 flags.go:52] FLAG: --alsologtostderr="false"
I0117 09:25:51.723750       1 flags.go:52] FLAG: --certs-dir="/etc/tls-certs"
I0117 09:25:51.723756       1 flags.go:52] FLAG: --httptest.serve=""
I0117 09:25:51.723760       1 flags.go:52] FLAG: --log-backtrace-at=":0"
I0117 09:25:51.723768       1 flags.go:52] FLAG: --log-dir=""
I0117 09:25:51.723773       1 flags.go:52] FLAG: --logtostderr="false"
I0117 09:25:51.723778       1 flags.go:52] FLAG: --stderrthreshold="0"
I0117 09:25:51.723783       1 flags.go:52] FLAG: --v="4"
I0117 09:25:51.723788       1 flags.go:52] FLAG: --vmodule=""
I0117 09:25:51.723796       1 main.go:52] Vertical Pod Autoscaler 0.3.0 Admission Controller
I0117 09:25:51.724091       1 certs.go:42] Successfully read 227 bytes from /etc/tls-certs/caKey.pem
I0117 09:25:51.724118       1 certs.go:42] Successfully read 558 bytes from /etc/tls-certs/caCert.pem
I0117 09:25:51.724135       1 certs.go:42] Successfully read 227 bytes from /etc/tls-certs/serverKey.pem
I0117 09:25:51.724154       1 certs.go:42] Successfully read 700 bytes from /etc/tls-certs/serverCert.pem
I0117 09:25:51.725221       1 reflector.go:131] Starting reflector *v1beta1.VerticalPodAutoscaler (1h0m0s) from k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/vpa/api.go:89
I0117 09:25:51.725258       1 reflector.go:169] Listing and watching *v1beta1.VerticalPodAutoscaler from k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/vpa/api.go:89
I0117 09:25:51.825093       1 shared_informer.go:123] caches populated
I0117 09:25:51.825127       1 api.go:93] Initial VPA synced successfully
I0117 09:25:51.831173       1 config.go:62] client-ca-file=-----BEGIN CERTIFICATE-----
MIIC7TCCAdWgAwIBAgIMFUAJhqR6oOjNCVIsMA0GCSqGSIb3DQEBCwUAMCIxIDAe
BgNVBAMTF2FwaXNlcnZlci1hZ2dyZWdhdG9yLWNhMB4XDTE4MDcwODE0NTIyN1oX
DTI4MDcwNzE0NTIyN1owIjEgMB4GA1UEAxMXYXBpc2VydmVyLWFnZ3JlZ2F0b3It
Y2EwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDMMmyD8Iuysy73IWXh
DUrEB+EEG0FrQFgBUeMsLCXDn/6fBGOcv/KvA5SUk+26cPD+4/YJdhNTpyxBdJyI
ZgeQROsntKFF+WU4uJ/eVUJkATvD/iycMlp+tKAyDob7MkEg0R506swXvDlPeie7
qpcW2hJlT98rJdoqECZUACVKcOQYplJhKTdcIStbAVwRrM4iY2hkWLRsiSuZPoFp
NJPPPQvTS8d1kzsnyAzJXPjmRDUFgSXbAKRmhfYAW5cJtrLw1W2XntQDbolqopPG
jlrazFgu4NtN++wN0ftMCHdnulV/9NhQUQQX8Qqo4gLgBbFyUPMj0wdL0MRof6Vc
B3n5AgMBAAGjIzAhMA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MA0G
CSqGSIb3DQEBCwUAA4IBAQCPXI8/0kNfl5xFBwqPuhdtDpzly1lpzCSAeDO3C7Bf
q8XWlgM0A/NczHhp09dbKPxQOkyMXelb55kyX0dp951bb93z1Qfp6RUthK2Utd5q
76WOV0w2wldiUeIxWR/zWk5zhr9jjBKPDCRVBMMc1PxJl4AIjp92GrsaRR0mNRRA
uaBMfXGYSfq9ksXT47Dkks3sPh0w8uaDgwrDE+uKlBqQoU/0Wa7MPvUUSODF8Qzx
0VZQdx7ilQKcHkcYtdg3h02JtLuVmGTdqDOs3YjMuqkkV79m5c6IgB7TJeYeC+Iy
yCuFAECjg54A/djgdj4xjjsEc5p4KfGjbskVC/3mPo/m
-----END CERTIFICATE-----
I0117 09:26:01.955043       1 config.go:131] Self registration as MutatingWebhook succeeded.
I0117 09:33:52.793027       1 reflector.go:357] k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/vpa/api.go:89: Watch close - *v1beta1.VerticalPodAutoscaler total 4 items received
I0117 09:43:34.799107       1 reflector.go:357] k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/vpa/api.go:89: Watch close - *v1beta1.VerticalPodAutoscaler total 10 items received
I0117 09:51:53.805197       1 reflector.go:357] k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/vpa/api.go:89: Watch close - *v1beta1.VerticalPodAutoscaler total 9 items received

```


And `kubectl describe mutatingWebhookConfiguration vpa-webhook-config`:

```
kubectl describe mutatingWebhookConfiguration vpa-webhook-config
Name:         vpa-webhook-config
Namespace:
Labels:       <none>
Annotations:  <none>
API Version:  admissionregistration.k8s.io/v1beta1
Kind:         MutatingWebhookConfiguration
Metadata:
  Creation Timestamp:  2019-01-17T09:26:01Z
  Generation:          1
  Resource Version:    32777004
  Self Link:           /apis/admissionregistration.k8s.io/v1beta1/mutatingwebhookconfigurations/vpa-webhook-config
  UID:                 e7f77da0-1a39-11e9-8af9-42010a840006
Webhooks:
  Client Config:
    Ca Bundle: ...
    Service:
      Name:        vpa-webhook
      Namespace:   kube-system
  Failure Policy:  Ignore
  Name:            vpa.k8s.io
  Namespace Selector:
  Rules:
    API Groups:

    API Versions:
      v1
    Operations:
      CREATE
    Resources:
      pods
    API Groups:
      autoscaling.k8s.io
    API Versions:
      v1beta1
    Operations:
      CREATE
      UPDATE
    Resources:
      verticalpodautoscalers
Events:  <none>
```

I think that this config is wrong:

```
Webhooks:
  Client Config:
    Ca Bundle: ...
    Service:
      Name:        vpa-webhook
      Namespace:   kube-system
```

I want to install VPA to another namespace. Can i do it?

**Comment by bskiba** [comment]
https://github.com/kubernetes/autoscaler/pull/1573 added support for configurable namespace for the webhook, but it is not included in 0.3.0 release. You can try to build your own VPA from head, or wait a bit and we can work to have this included in version 0.3.1
